papis/downloaders/tandfonline.py

```diff
@@ -22,41 +25,41 @@ def get_data(self):
         soup = self._get_soup()
         data.update(papis.downloaders.base.parse_meta_headers(soup))
 
-        doi = soup.find_all(name="meta",
-                attrs={"name": 'dc.Identifier', 'scheme': 'doi'})
-        if doi:
-            data['doi'] = doi[0].attrs.get('content')
-
         if 'author_list' in data:
             return data
 
         # Read brute force the authors from the source
         author_list = []
         authors = soup.find_all(name='span', attrs={'class': 'contribDegrees'})
-        cleanregex = re.compile(r'(^\s*|\s*$|&)')
-        editorregex = re.compile(r'([\n|]|\(Reviewing\s*Editor\))')
-        morespace = re.compile(r'\s+')
         for author in authors:
-            affspan = author.find_all('span', attrs={'class': 'overlay'})
-            afftext = affspan[0].text if affspan else ''
-            fullname = re.sub(',', '',
-                        cleanregex.sub('', author.text.replace(afftext, '')))
-            splitted = re.split(r'\s+', fullname)
-            cafftext = re.sub(' ,', ',',
-                              morespace.sub(' ', cleanregex.sub('', afftext)))
-            if 'Reviewing Editor' in fullname:
-                data['editor'] = cleanregex.sub(
-                    ' ', editorregex.sub('', fullname))
+            affiliation = author.find_all('span', attrs={'class': 'overlay'})
+            if affiliation:
+                # the span contains other things like the email, but we only
+                # want the starting text with the affiliation address
+                affiliation = next(affiliation[0].children)
+
+                affiliation = self.re_comma.sub(', ',
+                        self.re_clean.sub('', affiliation))
+                affiliation = [dict(name=affiliation)]
+
+            # find href="/author/escaped_fullname"
+            fullname = author.find_all('a', attrs={'class': 'entryAuthor'})
+            fullname = fullname[0].get('href').split('/')[-1]
+
+            fullname = re.sub(r'%2C', ',', re.sub(r'\+', ' ', fullname))
```

Comment:
I'm not that familiar with this sort of stuff, but is there a more generic way to replace these HTML codes (like `%2C`) with their unicode equivalent?